src/server.c

```diff
@@ -243,6 +243,69 @@ int user_max_connections(PgUser *user)
 	}
 }
 
+bool user_requires_auth_query(PgUser *user)
+{
+	/* users defined in auth_file with password do not require an auth_query */
+	return cf_auth_type != AUTH_TRUST && !user->from_auth_file;
+}
+
+void user_passwd_free(struct AANode *node, void *arg)
+{
+	PgUserPassword *user_passwd = container_of(node, PgUserPassword, tree_node);
+	free(user_passwd);
```

Comment:
`PgUserPassword` allocation/deallocation could be handled by slab allocation pooler if you think that's necessary.